### PR TITLE
[WIP] Add MiqAeService model for service template catalogs

### PR DESCRIPTION
**Pull request by fdupont-redhat**
Currently, it is not possible to explore service catalogs from Automate, while it is to explore service templates. This pull request closes the gap.

This is my first pull request ever, so please be gentle and guide me :)

### CODE REVIEW

File: lib/miq_automation_engine/service_models/miq_ae_service_service_template_catalog.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module MiqAeMethodService
+  class MiqAeServiceServiceTemplateCatalog < MiqAeServiceModelBase
+    expose :service_template_catalogs, :association => true
```
Comment (gmcculloug) [inline lib/miq_automation_engine/service_models/miq_ae_service_service_template_catalog.rb]: This is not an exposed relationship on the rails side: https://github.com/ManageIQ/manageiq/blob/master/app/models/service_template_catalog.rb#L6-L7

This line just needs to be removed.

File: spec/service_models/miq_ae_service_service_template_catalog_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module MiqAeServiceServiceTemplateCatalogSpec
+  describe MiqAeMethodService::MiqAeServiceServiceTemplateCatalog do
+    context "through an automation method" do
```
Comment (gmcculloug) [inline spec/service_models/miq_ae_service_service_template_catalog_spec.rb]: This context is not needed, should be removed.

Comment (fdupont-redhat) [inline spec/service_models/miq_ae_service_service_template_catalog_spec.rb]: Would you remove the whole block or just move the code out of it ?

File: lib/miq_automation_engine/service_models/miq_ae_service_service_template_catalog.rb (hunk 1 above)
Comment (fdupont-redhat) [inline lib/miq_automation_engine/service_models/miq_ae_service_service_template_catalog.rb]: Just to be sure, you talk about the line above ? The circular relationship ?

Comment (gmcculloug) [inline lib/miq_automation_engine/service_models/miq_ae_service_service_template_catalog.rb]: Yes, the `service_template_catalogs` relationship does not exist so the `expose` needs to be removed.

File: spec/service_models/miq_ae_service_service_template_catalog_spec.rb (hunk 2 above)
Comment (gmcculloug) [inline spec/service_models/miq_ae_service_service_template_catalog_spec.rb]: I meant the whole `context "through an automation method" do` block should be removed.

**Comment by fdupont-redhat** [comment]
I created a new PR, based on a branch, without the code from other PRs and cleaned from the non existent relationship and the useless test context. Here it is: https://github.com/ManageIQ/manageiq-automation_engine/pull/40.

**Comment by gmcculloug** [comment]
Closed in favor of PR https://github.com/ManageIQ/manageiq-automation_engine/pull/40

**Pull request closed without merging**